develop/develop-images/build_enhancements.md

```diff
@@ -151,3 +151,46 @@ $ docker build --no-cache --progress=plain --secret id=mysecret,src=mysecret.txt
 #9     duration: 1.470401133s
 ...
 ```
+
+## Using SSH to access private data in builds
+
+> **Acknowledgment**:
+> Special thanks to [Tonis Tiigi](https://medium.com/@tonistiigi) for granting
+> permission to use his blog post
+> [Build secrets and SSH forwarding in Docker 18.09](https://medium.com/@tonistiigi/build-secrets-and-ssh-forwarding-in-docker-18-09-ae8161d066)
+> as the basis of the content below.
+
+The `docker build` has a `--ssh` option to allow the Docker Engine to forward SSH agent connections. For more information 
+on SSH agent, see the [OpenSSH man page](https://man.openbsd.org/ssh-agent).
+
+Only the commands in the `Dockerfile` that have explicitly requested the SSH access by defining `type=ssh` mount have 
+access to SSH agent connections. The other commands have no knowledge of any SSH agent being available.
+
+To request SSH access for a `RUN` command in the `Dockerfile`, define a mount with type `ssh`. This will set up the 
+`SSH_AUTH_SOCK` environment variable to make programs relying on SSH automatically use that socket.
+
+Here is an example Dockerfile using SSH in the container:
+
+```
+# syntax=docker/dockerfile:experimental
```

Comment:
We may have to call out that the mount feature is not yet in the mainstream Dockerfile syntax (hence this `syntax=` being nescessary)

@tonistiigi ^^

Reply:
@tonistiigi could you please answer this? Thanks.

Reply:
Can add a line in the `Overriding default frontends` section that says: the new syntax features in Dockerfile are only available if you override the default frontend.

Reply:
Fixed